src/reactor/mod.rs

```diff
@@ -226,6 +227,18 @@ impl Core {
         res.expect("run should not return until future is done")
     }
 
+    /// Performs one iteration of the event loop, blocking on waiting for events
+    /// for at most `max_wait` (forever if `None`).
+    ///
+    /// It only makes sense to call this method if you've previously spawned
+    /// a future onto this event loop.
+    ///
+    /// `loop { lp.turn(None) }` is equivalent to calling `run` with an
+    /// empty future (one that never finishes).
+    pub fn turn(&mut self, max_wait: Option<Duration>) {
+        self.poll_internal(max_wait, &mut || true).expect("Error in event loop turn");
```

Comment:
Yes, but I'm not clear on the best way to do that. The duration argument would obviously need adding to `_run`, but how should `turn` indicate to `_run` that it must loop at least once? A `bool` argument?

Note that it requires a bit of care, because if the timeout expires without an event then finished will be false and `_run` will faithfully loop again.